Monitron.IM_RPC/ImRpc.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Monitron.ImRpc
+{
+    public class ImRpcCache  //q. needs to be singelton??
+    {
+        private static Dictionary<Type, Dictionary<string, MethodInfo>> MainCache = new Dictionary<Type, Dictionary<string, MethodInfo>>();
```

Comment:
Why do you have static caches? Just cache in the instance level. It's simpler and you'll have no collisions.